Source/Core/Core/IOS/ES/TitleManagement.cpp

```diff
@@ -644,39 +652,37 @@ IPCCommandResult ES::ExportContentBegin(Context& context, const IOCtlVRequest& r
 
 ReturnCode ES::ExportContentData(Context& context, u32 content_fd, u8* data, u32 data_size)
 {
-  const auto iterator = context.title_export.contents.find(content_fd);
-  if (!context.title_export.valid || iterator == context.title_export.contents.end() ||
-      iterator->second.content.m_position >= iterator->second.content.m_content.size)
+  ReturnCode ret = ES_EINVAL;
+  if (context.title_import.valid && context.title_import.content.valid && data && data_size != 0)
   {
-    return ES_EINVAL;
-  }
+    IOS::ES::Content content;
+    context.title_import.tmd.FindContentById(context.title_import.content.id, &content);
 
-  auto& metadata = iterator->second.content;
-
-  const auto& content_loader = AccessContentDevice(metadata.m_title_id);
-  const auto* content = content_loader.GetContentByID(metadata.m_content.id);
-  content->m_Data->Open();
+    std::vector<u8> buffer(data_size);
+    const s32 read_size =
+        ReadContent(content_fd, buffer.data(), static_cast<u32>(buffer.size()), 0);
+    if (read_size < 0)
+    {
+      CloseContent(content_fd, 0);
+      context.title_import = {};
+      return ES_SHORT_READ;
+    }
 
-  const u32 length =
-      std::min(static_cast<u32>(metadata.m_content.size - metadata.m_position), data_size);
-  std::vector<u8> buffer(length);
+    // IOS aligns the buffer to 32 bytes. Since we also need to align it to 16 bytes,
+    // let's just follow IOS here.
+    buffer.resize(Common::AlignUp(buffer.size(), 32));
 
-  if (!content->m_Data->GetRange(metadata.m_position, length, buffer.data()))
-  {
-    ERROR_LOG(IOS_ES, "ExportContentData: ES_SHORT_READ");
-    return ES_SHORT_READ;
+    const std::vector<u8> output =
+        Common::AES::Encrypt(context.title_import.key.data(),
+                             context.title_import.content.iv.data(), buffer.data(), buffer.size());
+    std::copy(output.cbegin(), output.cend(), data);
+    return ret;
```

Comment:
Oops, thanks for catching this! I've reorganised the function and removed `ret`.